contracts/examples/dao/DAO.sol

```diff
@@ -49,83 +57,61 @@ contract DAO is VentureEth {
     }
 
     /**
-     * @notice Propose funding for venture. Can only be used after the original funding round. venture must be of type IssuanceEth.
-     * @param funding The amount to proposed as funding for the venture.
-     * @param venture The proposed venture. Must be of type IssuanceEth.
+     * @notice Propose a venture. Must have approved the DAO to spend gage of your DAO tokens. venture must inherit from VentureEth.
+     * @param venture The address of the VentureEth contract
+     * @param funding The ether to fund the venture with.
      */
-    function proposeVenture(uint256 funding, address venture) public {
-        require(currentState == "LIVE", "DAO needs to be LIVE.");
-        uint256 newFundingPool = fundingPool.add(funding);
-        require(
-            newFundingPool <= address(this).balance,
-            "Not enough funds."
-        );
-        fundingPool = newFundingPool;
-        proposedFundingForVenture[address(VentureEth(venture))] = funding;
-    }
-
-    /**
-     * @notice Vote for venture. Can only be used after the original funding round.
-     * @param votes The amount of tokens to lock for the venture
-     * @param venture The venture to vote for. Must be of type VentureEth.
-     */
-    function voteForVenture(uint256 votes, address venture) public {
-        require(currentState == "LIVE", "DAO needs to be LIVE.");
-        require(
-            this.balanceOf(msg.sender).sub(totalVotesByHolder[msg.sender]) >= votes,
-            "Not enough power."
-        );
-        totalVotesForVenture[venture] = totalVotesForVenture[venture].add(
-            votes
+    function proposeVenture(
+        address venture,
+        uint256 funding
+    ) public {
+        require(currentState == "LIVE", "DAO needs to be LIVE");
+        this.transferFrom(msg.sender, address(this), gage);
+        Voting voting = new Voting(address(this), threshold);
+        voting.registerProposal(
+            address(this),
+            abi.encodeWithSignature("fundVenture(address,uint256)", venture, funding)
         );
-        totalVotesByHolder[msg.sender] = totalVotesByHolder[msg.sender].add(
-            votes
+        voting.registerProposal(
+            address(this),
+            abi.encodeWithSignature("retrieveVentureTokens(address)", venture)
         );
-        votesForVentureByHolder[venture][msg.sender] = votesForVentureByHolder[
-            venture][msg.sender].add(votes);
-        addBackerForVenture(venture, msg.sender);
+        voting.open();
+        proposals[venture] = address(voting);
+        ventures.add(venture);
```

Comment:
Funnily enough, we don't even need the `ventures` set. I'm sure that it would be useful for the front end, but it is not used anywhere.

For clarity, I don't mind keeping it. It is weird to not have a data structure with the ventures in the DAO portfolio. However I would add a venture to the `ventures` set only once its proposal is passed (so, in `fundVenture`) to avoid abusing the `ventures` set.